Source/InterceptorStrategies.cs

```diff
@@ -69,22 +69,26 @@ public InterceptionAction HandleIntercept(ICallContext invocation, InterceptorCo
 				return InterceptionAction.Continue;
 			}
 
-			IProxyCall lastMatchingSetup = null;
-			IProxyCall lastMatchingSetupForSameMethod = null;
+			IProxyCall matchedSetup = null;
 			foreach (var setup in ctx.OrderedCalls)
 			{
```

Comment:
Since `InterceptorContext.OrderedCalls`, now being a `Stack<IProxyCall>` instead of a `List<IProxyCall>`, is now in reverse order (last setup added gets enumerated first), we no longer search for the last matching setup, but for the first. This makes it possible to break out of the loop earlier.